File: src/1.Two Sum/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        map<int, int> M;
        int res;
        for(int i=0;i<nums.size();i++){
            res=target-nums[i];
            map<int, int>::iterator iter=M.find(res);
            if(iter!=M.end()){
                result.push_back(i);
                result.push_back(iter->second);
            }
            M[nums[i]]=i;
        }
        return result;
    }
};

File: src/1.Two Sum/Solution.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic = {}
        for i, element in enumerate(nums):
            res = target - element
            if res in dic:
                i1 = dic.get(res)
                i2 = i
                break
            else:
                dic[element] = i
        return (i1,i2)
